src/CanvasChart.js

```diff
@@ -347,16 +365,21 @@ export default function CanvasChart({
 
       const x = firstTask.month * monthWidth - MARGIN - LINE_SEGMENT_MIN_LENGTH;
 
+      // If this is a milestone, adjust the vertical line hieght accordingly.
+      // This math needs another look.
```

Comment:
Want to look now? 😄

Reply:
Tried to leave your core logic alone as much as possible, see comment